alf/trainers/policy_trainer.py

```diff
@@ -261,7 +261,12 @@ def train(self):
         self._save_checkpoint()
         if self._evaluate:
             self._eval_env.pyenv.close()
-        self._env.pyenv.close()
+        try:
+            # async off-policy driver has several envs, including self._env
+            for env in self._envs:
+                env.pyenv.close()
+        except:
+            self._env.pyenv.close()
```

Comment:
I feel the logic here is highly depending on the logic at off_policy_trainer, which is not good from the point of view of software engineering. We should have a cleaner expectation about self._env and self._envs in the base class  and the subclass follows those expectations.